extensions/ql-vscode/src/query-results.ts

```diff
@@ -45,45 +45,32 @@ export interface InitialQueryInfo {
 }
 
 export class CompletedQueryInfo implements QueryWithResults {
-  readonly query: QueryEvaluationInfo;
-  readonly message?: string;
-  readonly successful?: boolean;
-  /**
-   * The legacy result. This is only set when loading from the query history.
-   */
-  readonly result: legacyMessages.EvaluationResult;
-  readonly logFileLocation?: string;
-  resultCount: number;
-
-  /**
-   * Map from result set name to SortedResultSetInfo.
-   */
-  sortedResultsInfo: Record<string, SortedResultSetInfo>;
-
-  /**
-   * How we're currently sorting alerts. This is not mere interface
-   * state due to truncation; on re-sort, we want to read in the file
-   * again, sort it, and only ship off a reasonable number of results
-   * to the webview. Undefined means to use whatever order is in the
-   * sarif file.
-   */
-  interpretedResultsSortState: InterpretedResultsSortState | undefined;
-
-  /**
-   * Note that in the {@link readQueryHistoryFromFile} method, we create a CompletedQueryInfo instance
-   * by explicitly setting the prototype in order to avoid calling this constructor.
-   */
-  constructor(evaluation: QueryWithResults) {
-    this.query = evaluation.query;
-    this.logFileLocation = evaluation.logFileLocation;
-    this.result = evaluation.result;
-
-    this.message = evaluation.message;
-    this.successful = evaluation.successful;
-
-    this.sortedResultsInfo = {};
-    this.resultCount = 0;
-  }
+  constructor(
+    public query: QueryEvaluationInfo,
+
+    /**
+     * The legacy result. This is only set when loading from the query history.
+     */
+    public result: legacyMessages.EvaluationResult,
+    public logFileLocation?: string,
+    public successful?: boolean,
+    public message?: string,
```

Comment:
I think it would be better if we used `T | undefined` for the type instead of using `?` to mark the parameter as optional. At least for all the parameters up to `message` which don't have default values and used to come from the `QueryWithResults` object. This way when you call the constructor it'll force you to provide a value, even if you explicitly write `undefined`.
```suggestion
  constructor(
    public query: QueryEvaluationInfo,

    /**
     * The legacy result. This is only set when loading from the query history.
     */
    public result: legacyMessages.EvaluationResult,
    public logFileLocation: string | undefined,
    public successful: boolean | undefined,
    public message: string | undefined,
```

Reply:
I'm not entirely sure if this is so good.. here is the commit: https://github.com/github/vscode-codeql/pull/2242/commits/d5e3abf10891bda78434d66a264a2f1936e691b4
wdyt?

Reply:
I didn't explain too well so there's been a little confusion, but turns out it's all fine in the end.

I initially suggested changing the `CompletedQueryInfo` type rather than the `LocalQueryInfo` type. When constructing a `CompletedQueryInfo` we want to populate all of the fields from the `QueryWithResults`, but then some of the fields may be undefined. For `LocalQueryInfo` most of the arguments of the constructor are not needed at all in normal usage, and we only want them when deserializing query history.

I see that you updated the `CompletedQueryInfo` type anyway in https://github.com/github/vscode-codeql/pull/2242/commits/e3259bd99b4fbb15061548de16f4c78a6cf62230, so that's good. I don't think applying the same change the `LocalQueryInfo` is an improvement, so unless you disagree I'd suggest reverting https://github.com/github/vscode-codeql/commit/d5e3abf10891bda78434d66a264a2f1936e691b4.

Regarding the tests, I think they have been fixed in the background by Dave Bartolomeo's changes. By introducing the `completeThisQuery` function it's removed the call to the `CompletedQueryInfo` constructor from the tests, and it's that constructor call in the tests that wasn't correct and wasn't passing all the necessary arguments.

So I would recommend reverting https://github.com/github/vscode-codeql/commit/d5e3abf10891bda78434d66a264a2f1936e691b4, but then everything looks good 👍 

Does that make more sense?

Reply:
Yes, sounds good!